test/extensions/filters/http/grpc_json_transcoder/grpc_json_transcoder_integration_test.cc

```diff
@@ -64,8 +65,9 @@ class GrpcJsonTranscoderIntegrationTest
       response = codec_client_->makeHeaderOnlyRequest(request_headers);
     }
 
-    ASSERT_TRUE(fake_upstreams_[0]->waitForHttpConnection(*dispatcher_, fake_upstream_connection_));
-    if (!grpc_request_messages.empty()) {
+    if (!expected_grpc_request_messages.empty() || expected_upstream_request_body.length() != 0) {
```

Comment:
Diff is messy here. Only changes are:
- Line 68, 80, 98: Add conditions for new variable
- Rename `grpc_request_messages` -> `expected_grpc_request_messages`
- Rename `response_body` -> `expected_response_body`